aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -113,10 +113,7 @@ struct cpu_scatter_gather_base_kernel {
 
     dim = maybe_wrap_dim(dim, self.dim());
 
-    if (is_scatter_like) {
-      scatter_shape_check(self, dim, index, self);
-    }
-    else {
+    if (!is_scatter_like) {
       gather_shape_check(self, dim, index, self);
```

Comment:
I wonder why the gather shape checks couldn't get eliminated by structured kernels

Reply:
I guess this one specifically can be eliminated. The other one, however, cannot (line 222). That's because `gather` (not structured) is still using that code.